src/webgl/loading.js

```diff
@@ -437,6 +458,19 @@ function parseObj(model, lines, materials= {}) {
   if (model.vertexNormals.length === 0) {
     model.computeNormals();
   }
+  if (hasColoredVertices) {
+  // Ensure every vertex added has a color
+    if (coloredVerts.size !== uniqueVertices) {
+      throw new Error('Not all vertices have a color.');
```

Comment:
It would be great to add one more unit test with some invalid input to make sure it hits this error, then I think we're probably good to go!